Sally7/ValueConversion/FromS7Conversions.cs

```diff
@@ -113,13 +93,15 @@ private static void ConvertToByteArray<TTarget>(ref byte[]? value, in ReadOnlySp
         }
 
         private static void ConvertToBoolArray(ref bool[]? value, in ReadOnlySpan<byte> input, in int length)
-        {
-            value = new BitArray(input.Slice(0, (length + 7) / 8).ToArray()).Cast<bool>().Take(length).ToArray();
-        }
+            => value = new BitArray(input.Slice(0, (length + 7) / 8).ToArray()).Cast<bool>().Take(length).ToArray();
 
         private static void ConvertToString(ref string? value, in ReadOnlySpan<byte> input, in int length)
         {
+#if NETSTANDARD2_1_OR_GREATER
+            value = Encoding.ASCII.GetString(input.Slice(2, input[1]));
```

Comment:
We can avoid the allocation of the temporary array on NS 2.1 or newer (there won't be any newer anyway for .NET Standard as .NET 5 onwards is the future).